src/main.cpp

```diff
@@ -110,6 +189,38 @@ int main(int argc, char* argv[])
 
       return 0;
    }
+   if (app_options.user_string_dict().length() > 0) {
+      std::string path =
+         std::regex_replace(app_options.user_string_dict(), std::regex("\\\\"), "\\");
+      path = std::regex_replace(path, std::regex("\""), "");
+      if (fs::exists(path))
+         read_dictionary(path.c_str());
+      else {
+         std::cout << "Error: file '"s << path << "' does not exist\n";
+
+         return 0;
+      }
+   }
+   else
+      initialize_internal_dictionary();
+
+   std::string tmp = "";
+   int start_str = 0;
+   int end_str = 0;
+   
+   // add the input file names and filename + 'popular suffixes' to the hashes
+   for (int i = 0; i < input_files.size(); i++) {
+      start_str = input_files[i].find_last_of('\\');
+      if (start_str < 0) start_str = 0;
+      end_str = input_files[i].find_last_of('.');
+      tmp = input_files[i].substr(start_str, end_str - start_str);
+      add_hash(tmp);
+      add_hash("mapname.description." + tmp);
+      add_hash("mapname.name." + tmp);
+      for (int j = 0; j < popular_suffixes.size(); j++) {
+         add_hash(tmp + popular_suffixes[j]);
+      }
+   }
```

Comment:
This loop can probably be made easier to read by using range based for loops and [std::filesystem::path::stem](https://en.cppreference.com/w/cpp/filesystem/path/stem).

```suggestion
   for (const auto& input_file : input_files) {
      const auto name = fs::path{input_file}.stem().string();

      add_hash(name);
      add_hash("mapname.description."s += name);
      add_hash("mapname.name."s += name);

      for (const auto& suffix : common_layer_suffixes) {
         add_hash(std::string{name} += suffix);
      }
   }
```

As a final note both this and the code for loading the string dictionary would probably be better suited to be inside `extract_file` rather than `main`. Or even in their own function for `extract_file` to call.